server/controllers/student.py

```diff
@@ -89,13 +92,29 @@ def assignment(course, assign):
     assign = Assignment.by_name(assign, course.offering)
     if not assign:
         return abort(404)
+
     user_ids = assign.active_user_ids(current_user.id)
-    backups = assign.backups(user_ids).limit(5).all()
-    subms = assign.submissions(user_ids).limit(5).all()
-    final_submission = assign.final_submission(user_ids)
-    flagged = final_submission and final_submission.flagged
-    return render_template('student/assignment/index.html', course=course,
-            assignment=assign, backups=backups, subms=subms, flagged=flagged)
+    fs = assign.final_submission(user_ids)
+    group = Group.lookup(current_user, assign)
+    can_invite = assign.max_group_size > 1
+    can_remove = False
+    if group:
+        can_invite = len(group.members) < assign.max_group_size
+        can_remove = current_user in [m.user for m in group.members if m.status == 'active']
```

Comment:
All that's needed here is to figure out if the current user is actually a member. `user_ids` doesn't do this since it always has the current_users id.